pkg/app/middleware.go

```diff
@@ -64,3 +73,34 @@ func SetupCORSMiddleware(handler http.Handler, domains []string) http.Handler {
 	})
 	return corsHandler.Handler(handler)
 }
+
+func WithIPXEScriptMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			http.NotFound(w, r)
+			return
+		}
+		matches := ipxeScriptPattern.FindStringSubmatch(r.URL.Path)
+		if matches == nil {
+			// Path doesn't match the regexp
+			http.NotFound(w, r)
+			return
+		}
+		if _, err := uuid.Parse(matches[1]); err != nil {
+			// Invalid UUID format
+			http.NotFound(w, r)
+			return
+		}
```

Comment:
Same comment as the code in the image service, for the purpose of this middleware, we don't care about this so I don't think we need to check it here.

Reply:
Right ,synced with proposed image-service change